### PR TITLE
[ADZ-143] API Spec auto imported from Apigee

### PR DESCRIPTION
**Pull request by ziemsky**
This is a preliminary pull request. As discussed, the entire `pds_api_spec_epic` branch is a long running feature branch which will not be merged into master for quite some time.

This pull request is one of a number of partial ones, so that we avoid a humongous pull request at the end.

Final merge (into `master`) will be covered by a separate pull request, by which time most of the code will be well known to the reviewer doing the partial reviews in the meantime.

Note that classes around communication with Apigee (`ApigeeService` and its dependencies) are candidate for a major rework as described by [ADZ-127](https://jira.digital.nhs.uk/browse/ADZ-127); as such, they are not refined or sufficiently covered with tests, yet. This will be addressed by ADZ-127 which will be implemented before the merge to master.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -256,6 +275,54 @@
...
+                <groupId>io.swagger</groupId>
+                <artifactId>swagger-core</artifactId>
+                <version>1.6.1</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.slf4j</groupId>
+                <artifactId>slf4j-ext</artifactId>
+                <version>1.7.30</version>
+            </dependency>
```
Comment (GeoffHayward) [inline pom.xml:293]: Is this supplied by the BR parent POM? I worry about version conflicts.

File: pom.xml [hunk 2]
```diff
@@ -256,6 +275,54 @@
...
+                <artifactId>jackson-coreutils</artifactId>
+                <version>1.8</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.eclipse.jetty</groupId>
+                <artifactId>jetty-server</artifactId>
+                <version>${org.eclipse.jetty.version}</version>
+                <scope>test</scope>
+            </dependency>
```
Comment (GeoffHayward) [inline pom.xml:306]: What does Jetty do in the project?

File: cms/src/main/java/uk/nhs/digital/apispecs/jcr/ApiSpecificationDocumentJcrRepository.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import javax.jcr.query.Query;
+import javax.jcr.query.QueryResult;
+
+public class ApiSpecificationDocumentJcrRepository implements ApiSpecificationDocumentRepository {
+
+    private static final Logger log = LoggerFactory.getLogger(ApiSpecificationDocumentJcrRepository.class);
+
+    private static final String WEBSITE_DOCS_FOLDER_JCR_PATH = "/jcr:root/content/documents/corporate-website";
+    private static final String QUERY_STATEMENT =
+        WEBSITE_DOCS_FOLDER_JCR_PATH + "//element(*, website:apispecification)/..[@jcr:primaryType='hippo:handle']";
```
[the remark this answers is not in the document]
Comment (ziemsky) [reply cms/src/main/java/uk/nhs/digital/apispecs/jcr/ApiSpecificationDocumentJcrRepository.java:25]: Good idea - it gives a name to the component. I'll apply the change before merging.

File: pom.xml (hunk 1 above)
Comment (ziemsky) [reply pom.xml:293]: Good concern :+1: 

Short answer is: it's not - it's a transitive dependency, first introduced by the `content-exim-core` module. Swagger CodeGen library introduces it as a transitive dependency as well, but with a slightly different version.

That version inconsistency was flagged by `maven-enforcer-plugin` which showed all the (different) transitively included versions and what requires them. I specified this dependency in the `dependencyManagement` section of the POM file to explicitly 'fix' it at the highest version of the 1.7.x line that was mentioned in the Maven Enforcer's report.

Luckily, as shown by the report below, all versions of the those dependencies differ only in their 'patch' segment so I don't expect any incompatibilities:

```
[WARNING] Rule 2: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.slf4j:slf4j-ext:1.7.12 paths to dependency are:
+-uk.nhs.digital:website-cms:dev-SNAPSHOT
  +-io.swagger.codegen.v3:swagger-codegen-generators:1.0.19
    +-org.slf4j:slf4j-ext:1.7.12
and
+-uk.nhs.digital:website-cms:dev-SNAPSHOT
  +-org.onehippo.forge.content-exim:content-exim-core:3.0.1
    +-org.slf4j:slf4j-ext:1.7.25
and
+-uk.nhs.digital:website-cms:dev-SNAPSHOT
  +-io.swagger.codegen.v3:swagger-codegen-generators:1.0.19
    +-io.swagger.codegen.v3:swagger-codegen:3.0.19
      +-org.slf4j:slf4j-ext:1.7.25
and
+-uk.nhs.digital:website-cms:dev-SNAPSHOT
  +-org.onehippo.cms7:hippo-cms-test:13.4.2
    +-org.onehippo.cms7:hippo-repository-testutils:13.4.2
      +-org.slf4j:slf4j-ext:1.7.25
and
+-uk.nhs.digital:website-cms:dev-SNAPSHOT
  +-io.swagger.codegen.v3:swagger-codegen-generators:1.0.19
    +-io.swagger.codegen.v3:swagger-codegen:3.0.19
      +-io.swagger:swagger-parser:1.0.50
        +-org.slf4j:slf4j-ext:1.7.28
and
+-uk.nhs.digital:website-cms:dev-SNAPSHOT
  +-io.swagger.codegen.v3:swagger-codegen-generators:1.0.19
    +-io.swagger.codegen.v3:swagger-codegen:3.0.19
      +-io.swagger.parser.v3:swagger-parser:2.0.19
        +-org.slf4j:slf4j-ext:1.7.30
and
+-uk.nhs.digital:website-cms:dev-SNAPSHOT
  +-io.swagger.codegen.v3:swagger-codegen-generators:1.0.19
    +-io.swagger.codegen.v3:swagger-codegen:3.0.19
      +-io.swagger:swagger-codegen:2.4.13
        +-org.slf4j:slf4j-ext:1.7.12
```

File: pom.xml (hunk 2 above)
Comment (ziemsky) [reply pom.xml:306]: Jetty is only transitively included by a couple of libraries used exclusively in testing:
* hippo-cms-test
* wiremock-jre8

I declared it explicitly in the `dependencyManagement` section for the same reasons as those described for `slf4j-ext` above - various versions of the dependency were tripping Maven Enforcer and were increasing the risk of dependency conflicts.

**Pull request merged**